core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -148,6 +156,55 @@ public Expression visitNot(Not node, AnalysisContext context) {
     return dsl.not(node.getExpression().accept(this, context));
   }
 
+  @Override
+  public Expression visitRelevanceFunction(RelevanceFunction node, AnalysisContext context) {
+    FunctionName functionName = FunctionName.of(node.getFuncName());
+    TypeEnvironment typeEnv = context.peek();
+
+    List<Expression> arguments =
+        node.getFuncArgs().stream()
+            .map(unresolvedExpression -> analyze(unresolvedExpression, context))
+            .collect(Collectors.toList());
+
+    // Ensure each field is valid
+    switch(node.getType()) {
+      case NO_FIELD_FUNCTION:
+        Predicate<String> filter_predicate = Pattern
```

Comment:
this will only get the first field.  It's possible to have multiple fields like this: 
```
"field1:query_expr AND field2:query_expr OR field3:query_expr OR ..."
```